Sources/AsyncHTTPClient/NIOTransportServices/TLSConfiguration.swift

```diff
@@ -110,29 +110,55 @@
             // renegotiation support key is unsupported
 
             // trust roots
-            if let trustRoots = self.trustRoots {
-                guard case .default = trustRoots else {
-                    preconditionFailure("TLSConfiguration.trustRoots != .default is not supported")
+            var secTrustRoots: [SecCertificate]?
+            switch trustRoots {
+            case .some(.certificates(let certificates)):
+                secTrustRoots = try certificates.compactMap { certificate in
+                    try SecCertificateCreateWithData(nil, Data(certificate.toDERBytes()) as CFData)
                 }
+            case .some(.file):
+                preconditionFailure("TLSConfiguration.trustRoots.file is not supported")
```

Comment:
same q here. @Lukasa wdyt?

Reply:
This is a slightly different case. I could probably support this case if I knew what format the file was in.

Reply:
The file format for trust roots is just bundle-o-pem: a collection of PEM-encoded X509 certs in the same file.

Reply:
Now loading PEM info `NIOSSLCertificate` and generating DER for loading into `SecCertificateCreateWithData`. Possibly not the quickest route.